ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/perception/RDXStancePoseSelectionPanel.java

```diff
@@ -65,35 +84,25 @@ public RDXStancePoseSelectionPanel(StancePoseCalculator stancePoseCalculator)
       pickPointSphere = RDXModelBuilder.createSphere(0.04f, Color.CYAN);
    }
 
-   public void update(SideDependentList<FramePose3D> goalPoses, TerrainMapData terrainMapData, HeightMapData heightMapData)
+   public void update(TerrainMapData terrainMapData, HeightMapData heightMapData)
    {
       environmentHandler.setHeightMap(heightMapData);
       environmentHandler.setTerrainMapData(terrainMapData);
-
-      boolean panel3DIsHovered = latestInput != null && latestInput.isWindowHovered();
-      if (panel3DIsHovered && ImGui.isKeyPressed('P'))
-      {
-         selectionActive = true;
-      }
-      if (ImGui.isKeyPressed(ImGuiTools.getEscapeKey()))
-      {
-         selectionActive = false;
-      }
+      
       updatePoses();
    }
 
    private void updatePoses()
    {
-      if (environmentHandler.hasTerrainMapData())
+      if (environmentHandler.hasTerrainMapData() && environmentHandler.hasHeightMap())
```

Comment:
is there ever a case where you won't have height map data if you have terrain map data? should this be an error thrown in the get height in world method?

Reply:
I'm not sure, in this case where `updatePoses()` is called we are setting both the terrain map and the height map so I think it makes sense to check that its not null